src/tools/error_index_generator/main.rs

```diff
@@ -159,18 +160,27 @@ This page lists all the error codes emitted by the Rust compiler.
         <title>Rust error codes index - Error codes index</title>
         <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
         <meta name="description" content="Book listing all Rust error codes">
-        <script src="error_codes/redirect.js"></script>
+        <script>
+            (function() {
+                if (window.location.hash) {
+                    let code = window.location.hash.replace(/^#/, '');
+                    // We have to make sure this pattern matches to avoid
+                    // inadvertently creating an open redirect.
+                    if (/^E[0-9]+$/.test(code)) {
+                        window.location.replace('./error_codes/' + code + '.html');
```

Comment:
Why not making this change in the JS file directly? It had the advantage of keeping things split (JS split from HTML) which is easier to read/maintain.

Reply:
Oh, no particular reason. It was no longer being shared between different pages, and it seemed a little more convenient to have everything in one place. I don't have a strong preference, though, so I pushed an update to keep it in the old location.

Reply:
Thanks! It also makes the changes much smaller. :)